src/graphql/data/util.ts

```diff
@@ -0,0 +1,49 @@
+import { useWeb3React } from '@web3-react/core'
+import { SupportedChainId } from 'constants/chains'
+
+import { Chain } from './__generated__/TokenQuery.graphql'
+import { HistoryDuration } from './__generated__/TokenTopProjectsQuery.graphql'
+
+export enum TimePeriod {
+  HOUR,
+  DAY,
+  WEEK,
+  MONTH,
+  YEAR,
+  ALL,
+}
+
+export function toHistoryDuration(timePeriod: TimePeriod): HistoryDuration {
+  switch (timePeriod) {
+    case TimePeriod.HOUR:
+      return 'HOUR'
+    case TimePeriod.DAY:
+      return 'DAY'
+    case TimePeriod.WEEK:
+      return 'WEEK'
+    case TimePeriod.MONTH:
+      return 'MONTH'
+    case TimePeriod.YEAR:
+      return 'YEAR'
+    case TimePeriod.ALL:
+      return 'MAX'
+  }
+}
+
+export const CHAIN_IDS_TO_BACKEND_NAME: { [key: number]: Chain } = {
+  [SupportedChainId.MAINNET]: 'ETHEREUM',
+  [SupportedChainId.GOERLI]: 'ETHEREUM_GOERLI',
+  [SupportedChainId.POLYGON]: 'POLYGON',
+  [SupportedChainId.POLYGON_MUMBAI]: 'POLYGON',
+  [SupportedChainId.CELO]: 'CELO',
+  [SupportedChainId.CELO_ALFAJORES]: 'CELO',
+  [SupportedChainId.ARBITRUM_ONE]: 'ARBITRUM',
+  [SupportedChainId.ARBITRUM_RINKEBY]: 'ARBITRUM',
+  [SupportedChainId.OPTIMISM]: 'OPTIMISM',
+  [SupportedChainId.OPTIMISTIC_KOVAN]: 'OPTIMISM',
+}
+
+export function useCurrentChainName() {
+  const { chainId } = useWeb3React()
+  return chainId ? CHAIN_IDS_TO_BACKEND_NAME[chainId] : 'ETHEREUM'
```

Comment:
I think this will crash the app when the connected chain isn't supported by the protocol (but specifically by CHAIN_IDS_TO_BACKEND_NAME)

Reply:
fixed